scripts/jenkins/ardana/ansible/deploy-caasp.yml

```diff
@@ -40,13 +39,29 @@
     always_run: True
     shell: |
       source ~ardana/service.osrc
-      openstack stack create --wait --timeout 120 -t caasp-stack.yaml -e caasp-environment.yaml --parameter image="{{ image_name }}" "{{ stack_name }}"
-      openstack stack show "{{ stack_name }}"
-      status=$(openstack stack show "{{ stack_name }}" | grep -w stack_status | awk '{print $4}')
+      openstack stack create --wait --timeout 120 -t caasp-stack.yaml -e caasp-environment.yaml --parameter image="{{ image_name }}" --parameter admin_flavor=m1.medium --parameter master_flavor=m1.medium --parameter worker_flavor=m1.medium "caasp-stack"
+      openstack stack show "caasp-stack"
+      status=$(openstack stack show "caasp-stack" | grep -w stack_status | awk '{print $4}')
       if [ $status = 'CREATE_COMPLETE' ]; then echo "CaaSP stack created successfully."; else echo "Failed to create stack: $status"; exit 1; fi
     args:
       executable: /bin/bash
       chdir: /usr/share/caasp-openstack-heat-templates/
+    when:
+      - deploy_caasp == 'single-master'
+
+  - name: Create heat stack for multi master CaaSP deployment
```

Comment:
There's a fair amount of duplication in here compared to the next task.

This would look a lot cleaner if you had an ansible role, where you can keep differentiated variable files that you can load conditionally, then use those variables in tasks, or even model this long command as a bash template that you can generate then run on the deployer node. 

Either way, my point is this playbook is getting to be complex enough that an ansible role could help better manage that complexity. You can do that here, or in a follow-up PR, depending on how urgent this needs to be available.

Reply:
Thanks @stefannica I tried to address your concerns about code duplication by setting appropriate parameters using "set_fact". It does fall short of the creating new role but might be good starting point for the future.  Can you please take a look?  This is still under test.

BTW: I am having  a hard time finding a suitable ardana model or scenario which will be adequate to create 3 or 4, x1.large instances. Was there a model which we used in the past for earlier version of this CI test? I tried using std-3cm but that also wasn't sufficient.

Reply:
@aagate unfortunately, the CI doesn't yet support controlling the ECP flavors for the virtual compute nodes. The flavors come from the [input model generator](https://github.com/SUSE-Cloud/automation/tree/master/scripts/jenkins/ardana/ansible/roles/input_model_generator) scenario ([here](https://github.com/SUSE-Cloud/automation/blob/master/scripts/jenkins/ardana/ansible/roles/input_model_generator/vars/templates/service/standard.yml#L51) for example). 

If you need bigger flavors, you either have to create a new input model generator scenario - which I don't recommend, because then you'll only be able to deploy CaaSP on that one scenario - or improve the input model generator feature to have a parameterized compute node flavor.